### PR TITLE
fixed auto completion not suggest const for array

### PR DESCRIPTION
**Pull request by msivasubramaniaan**
### What does this PR do?

Auto completion should not suggest const value from schema for Array

### What issues does this PR fix or reference?

https://github.com/redhat-developer/yaml-language-server/issues/620

### Is it tested? How?

Yes. with UT

### CODE REVIEW

File: src/languageservice/services/yamlCompletion.ts [hunk 1]
```diff
@@ -663,7 +663,7 @@ export class YamlCompletion {
         //    - item1
         // it will treated as a property key since `:` has been appended
         if (nodeParent && isSeq(nodeParent) && schema.schema.type !== 'object') {
-          this.addSchemaValueCompletions(schema.schema, separatorAfter, collector, {});
+          this.addSchemaValueCompletions(schema.schema, separatorAfter, collector, {}, isArray);
```
Comment (gorkem) [inline src/languageservice/services/yamlCompletion.ts:666]: If this is the only place `isArray` is used can we delay its initialization to inside this `if` block?

**Review by gorkem — CHANGES_REQUESTED** [review]

File: src/languageservice/services/yamlCompletion.ts [hunk 2]
```diff
@@ -663,7 +662,8 @@ export class YamlCompletion {
         //    - item1
         // it will treated as a property key since `:` has been appended
         if (nodeParent && isSeq(nodeParent) && schema.schema.type !== 'object') {
-          this.addSchemaValueCompletions(schema.schema, separatorAfter, collector, {});
+          const isArray = nodeParent.items.length > 0;
```
Comment (evidolob) [inline src/languageservice/services/yamlCompletion.ts:665]: Better to use `Array.isArray()` to check is `nodeParent.items` array

**Review by evidolob — COMMENTED** [review]
LGTM

**Review by evidolob — APPROVED** [review]
LGTM

**Review by gorkem — APPROVED** [review]
lgtm

**Pull request merged**